### PR TITLE
Refs #28538 - Drop smart_variable left overs

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
@tbrisker this refs an already closed issue. Do we want to cherry pick this into 2.0 or should I open a new one?

File: app/controllers/concerns/api/v2/lookup_keys_common_controller.rb [hunk 1]
```diff
@@ -106,33 +77,22 @@ def smart_class_parameters_resource_scope
...
+    @smart = @smart_class_parameter
   end
 
   def return_if_smart_mismatch
     if (@smarts && @smart && !@smarts.find_by_id(@smart.id)) || (@smarts && !@smart)
-      obj = smart_variable_id? ? "Smart variable" : "Smart class parameter"
-      id = if smart_variable_id?
-             params.key?('smart_variable_id') ? params['smart_variable_id'] : params['id']
-           else
-             params.key?('smart_class_parameter_id') ? params['smart_variable_id'] : params['id']
```
Comment (ekohl) [inline app/controllers/concerns/api/v2/lookup_keys_common_controller.rb:126]: Note this previously appeared to have a bug. Note the mismatch.

**Review by tbrisker — COMMENTED** [review]
The previous pr was a continuation of the permission cleanup.
The full cleanup effort is being done in steps on purpose and tracked under https://projects.theforeman.org/issues/28278, to make it easier to review in parts finish rather than one giant PR removing everything.

**Comment by tbrisker** [comment]
Closing for now, this whole concern will be refactored/dropped in https://projects.theforeman.org/issues/28596.

**Pull request closed without merging**